### PR TITLE
Jmps java 9

### PR DESCRIPTION
**Pull request by Warkdev**
Adding JPMS module définition to picocli.
Adding new subproject picocli-core-module

### CODE REVIEW

File: picocli-core-module/build.gradle [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    }
+    into "${projectDir}/src/main/java"
+}
+
+compileJava {
+    inputs.property("moduleName", moduleName)   
+    doFirst {
+        options.compilerArgs = [
+            '--module-path', classpath.asPath,
+        ]
```
Comment (remkop) [inline picocli-core-module/build.gradle]: Looks like this fails the build on Java 8... I see this:

```
> Task :picocli-core-module:compileJava FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':picocli-core-module:compileJava'.
> invalid flag: --module-path
```

Maybe we can work around that with something like this:

```
doFirst {
    if (org.gradle.api.JavaVersion.current().isJava9Compatible()) {
        options.compilerArgs = [
            '--module-path', classpath.asPath,
        ]
    }
    classpath = files()
}

**Comment by remkop** [comment]
I went ahead and merged the PR. 
Thank you for the hard work!
With the version check the build now passes on Java 8, so it's looking good!

**Comment by remkop** [comment]
Link to the original issue:
https://github.com/remkop/picocli/issues/495

File: picocli-core-module/build.gradle (hunk 1 above)
Comment (jwheeler91) [reply picocli-core-module/build.gradle]: Java 8 and below won't understand the --module-path argument so that makes sense to me...

Comment (jwheeler91) [reply picocli-core-module/build.gradle]: If this core module is aimed squarely at the JPMS features, why do we need to support java 8 anyway as it the main jar will support java 8?

Comment (remkop) [reply picocli-core-module/build.gradle]: The plugin that’s used allows us to build a Java 9 module with just Java 8. It uses a byte code manipulation library for this. That makes it easier for contributors to set up the environment and build the project so I like that.

Comment (jwheeler91) [reply picocli-core-module/build.gradle]: Cool. I see. Nice work

Comment (Warkdev) [reply picocli-core-module/build.gradle]: @jwheeler, I'm not in this weekend, feel free to update the gradle file as remkop suggested. 😊

**Comment by remkop** [comment]
I went ahead and merged the PR, and already made the above changes. I'll put further comments on #495.

**Pull request merged**